src/plugins/discover/public/application/main/components/log_explorer/log_explorer_main_route.tsx

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useEffect, useState, memo, useCallback } from 'react';
+import { useParams, useHistory } from 'react-router-dom';
+import { SavedObject } from '@kbn/data-plugin/public';
+import { ISearchSource } from '@kbn/data-plugin/public';
+import {
+  DataViewAttributes,
+  DataViewSavedObjectConflictError,
+} from '@kbn/data-views-plugin/public';
+import { redirectWhenMissing } from '@kbn/kibana-utils-plugin/public';
+import { useExecutionContext } from '@kbn/kibana-react-plugin/public';
+import {
+  AnalyticsNoDataPageKibanaProvider,
+  AnalyticsNoDataPage,
+} from '@kbn/shared-ux-page-analytics-no-data';
+import {
+  SavedSearch,
+  getSavedSearch,
+  getSavedSearchFullPathUrl,
+} from '../../../../services/saved_searches';
+import { getState } from '../../services/discover_state';
+import { loadIndexPattern, resolveIndexPattern } from '../../utils/resolve_index_pattern';
+import { LogExplorerMainApp } from './log_explorer_main_app';
+import { getRootBreadcrumbs, getSavedSearchBreadcrumbs } from '../../../../utils/breadcrumbs';
+import { LoadingIndicator } from '../../../../components/common/loading_indicator';
+import { DiscoverError } from '../../../../components/common/error_alert';
+import { useDiscoverServices } from '../../../../hooks/use_discover_services';
+import { getUrlTracker } from '../../../../kibana_services';
+
+const LogExplorerMainAppMemoized = memo(LogExplorerMainApp);
+
+interface DiscoverLandingParams {
+  id: string;
+}
+
+interface Props {
+  isDev: boolean;
+}
+
+export function DiscoverLogExplorerRoute(props: Props) {
+  const history = useHistory();
+  const services = useDiscoverServices();
+  const { isDev } = props;
+  const {
+    core,
+    chrome,
+    uiSettings: config,
+    data,
+    toastNotifications,
+    http: { basePath },
+    dataViewEditor,
+  } = services;
+  const [error, setError] = useState<Error>();
+  const [savedSearch, setSavedSearch] = useState<SavedSearch>();
+  const indexPattern = savedSearch?.searchSource?.getField('index');
+  const [indexPatternList, setIndexPatternList] = useState<Array<SavedObject<DataViewAttributes>>>(
+    []
+  );
+  const [hasESData, setHasESData] = useState(false);
+  const [hasUserDataView, setHasUserDataView] = useState(false);
+  const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
+  const { id } = useParams<DiscoverLandingParams>();
+
+  useExecutionContext(core.executionContext, {
+    type: 'application',
+    page: 'app',
+    id: id || 'new',
+  });
+
+  const loadDefaultOrCurrentIndexPattern = useCallback(
```

Comment:
We don't really need to change this, as it works fine, and we want the exact same functionality. But, this is a bit of a loaded function, it does do what it says, but it also performs a lot of other logic.

Reply:
One thing is the logic to detect the no data state. this can stay as it's similar in other places. other logic is about the saved search and data view(s). this should be refactored (It's a leftover how it worked in the Angular world)